mlcomp/parallelm/mlpiper/wizard_shell.py

```diff
@@ -0,0 +1,383 @@
+        except CommandShortCircuit as e:
+            return e.code
+
+        cmd = self.SUBCOMMAND_ADD
+        self._print_header = shell._wizard_edit_mode
+        if args is not None and len(args):
+            if args[0] not in self._handlers.keys():
+                self.usage_error(shell, args)
+                return 1
+
+            cmd = args[0]
+
+        return self._handlers[cmd](shell, args)
+
+    def _add_handler(self, shell, args=None):
+        component_info = shell._component_info
+
+        comp_arg = ComponentArgumentInfo()
+        w = ArgumentWizard(component_info, comp_arg)
+        ns = w.run(shell, self._print_header)
+        if ns:
+            dd = {}
+            for k, v in ns.__dict__.items():
+                if isinstance(v, six.string_types) and len(v) == 0:
+                    continue
+                dd[k] = v
+
+            comp_arg.load_from_json(dd)
+            if component_info.arguments is None:
+                component_info.arguments = []
+            component_info.arguments.append(comp_arg)
+        return 0
+
+    def _edit_handler(self, shell, args):
+        if len(args) < 2:
+            self.usage_error(shell, args)
+            return 1
+        component_info = shell._component_info
+        key = args[1]
+        comp_arg = component_info.get_argument(key)
+        if comp_arg:
+            w = ArgumentWizard(component_info, comp_arg)
+            ns = w.run(shell, self._print_header)
+            if ns:
+                dd = {}
+                for k, v in ns.__dict__.items():
+                    if isinstance(v, six.string_types) and len(v) == 0:
+                        continue
+                    dd[k] = v
+
+                comp_arg.load_from_json(dd)
+        else:
+            print("Argument with the key '{}' is not found".format(key))
+            self.usage_error(shell, args)
+
+        return 0
+
+    def _delete_handler(self, shell, args):
+        if len(args) < 2:
+            self.usage_error(shell, args)
+            return 1
+        component_info = shell._component_info
+        key = args[1]
+        arg = component_info.get_argument(key)
+        if arg:
+            component_info.arguments.remove(arg)
+        else:
+            print("Argument with the key '{}' is not found".format(key))
+            self.usage_error(shell, args)
+        return 0
+
+
+class ShowCommand(Command):
+
+    def __init__(self, name, topic, brief, **kwargs):
+
+        self.parser = PypsiArgParser(prog=name, description=brief)
+
+        super(ShowCommand, self).__init__(name=name, usage=self.parser.format_help(), topic=topic, brief=brief, **kwargs)
+
+    def run(self, shell, args):
+        try:
+            ns = self.parser.parse_args(args)
+        except CommandShortCircuit as e:
+            return e.code
+
+        component_info = shell._component_info
+        print(json.dumps(component_info, indent=4, cls=ComponentInfoEncoder))
```

Comment:
Why not using `pprint` module?

Reply:
I don't think it is useful here.
Component info is an instance of ComponentInfo class which is not fundamental Python class.

Also ComponentInfoEncoder uses OrderedDict to preserve fields order.